src/Lecture2.hs

```diff
@@ -194,7 +362,12 @@ verify that.
 [1,2,3,4,7]
 -}
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge xs [] = xs
+merge [] ys = ys
+merge (x:xs) (y:ys)
+  | x == y    = [x, y] ++ merge xs ys
```

Comment:
Generally it's better to use `:` instead of `++` when appending to the beginning 🙂 
```suggestion
  | x == y    = x : y : merge xs ys
```